src/qibo/backends/__init__.py

```diff
@@ -78,39 +78,73 @@ class GlobalBackend(NumpyBackend):
         {"backend": "pytorch"},
     ]
 
-    def __new__(cls):
-        if cls._instance is not None:
-            return cls._instance
+    @classmethod
+    def backend(cls):  # pragma: no cover
+        if cls._backend is not None:
+            return cls._backend
 
         backend = os.environ.get("QIBO_BACKEND")
-        if backend:  # pragma: no cover
+        if backend:
             # Create backend specified by user
             platform = os.environ.get("QIBO_PLATFORM")
-            cls._instance = construct_backend(backend, platform=platform)
+            cls._backend = construct_backend(backend, platform=platform)
         else:
             # Create backend according to default order
             for kwargs in cls._default_order:
                 try:
-                    cls._instance = construct_backend(**kwargs)
+                    cls._backend = construct_backend(**kwargs)
                     break
-                except (ImportError, MissingBackend):
+                except (ModuleNotFoundError, ImportError):
                     pass
 
-        if cls._instance is None:  # pragma: no cover
+        if cls._backend is None:
             raise_error(RuntimeError, "No backends available.")
 
-        log.info(f"Using {cls._instance} backend on {cls._instance.device}")
-        return cls._instance
+        log.info(f"Using {cls._backend} backend on {cls._backend.device}")
+        return cls._backend
 
     @classmethod
     def set_backend(cls, backend, **kwargs):  # pragma: no cover
         if (
-            cls._instance is None
-            or cls._instance.name != backend
-            or cls._instance.platform != kwargs.get("platform")
+            cls._backend is None
```

Comment:
Maybe we could drop this case. Calling `set_backend(None)` does not seem that valuable, and if you want to set the default backend in principle you should do nothing, since the `backend` class method is already returning it whenever you request a backend without setting anything.

Moreover:
> There should be one-- and preferably only one --obvious way to do it.
> https://peps.python.org/pep-0020/

Reply:
Updated this.